src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -156,10 +141,35 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
         End Sub
 
         Private Structure FormattingInfo
+            ' This is the original flag to enable semantic formating on a span
             Public UseSemantics As Boolean
+            ' This is new flag that allows semantic formating of a single statement
+            Public UseSemanticStatementFormatting As Boolean
+            ' This is the total span to format, the span may or may not be semantically formated
             Public SpanToFormat As SnapshotSpan
```

Comment:
why not just have this be SpanToSyntacticallyFormat (and also be nullable).  Then, teh consumer of this just looks at both, sees which are not null, and formats each one appropriately.

Reply:
For SpanToSyntacticallyFormat no reason except I don't do a lot with nullable.